### PR TITLE
Lists of contents as viewed by ghost now looks a lot nicer

### PR DESCRIPTION
**Pull request by ghost**
This uses the \a and \s macros for listing singular and plural items, with the caveat of \s not being grammatically correct all the time (though \a seemingly is)

This makes a list of all unique item types in a list, counts items of each type, and displays them numerated as opposed to just every single item no matter how many times it gets repeated.

Example: A list that would normally say "It contains: the multitool, the multitool, the jackboots, the jackboots and the oxygen tank" would now say something like "It contains: 2 multitools, 2 jackbootss and a oxygen tank" (or "an oxygen tank", hopefully)
[tweak][grammar]

:cl:
 * tweak: Certain item lists now display a much nicer, counted list of items when examined as a ghost

### CODE REVIEW

File: code/__HELPERS/lists.dm [hunk 1]
```diff
@@ -11,24 +11,38 @@
...
+	var/typecount = 0
+	var/currentName = ""
+	if (!uniquetotal)
 		return "[nothing_text]"
-	else if (total == 1)
-		return "[input[1]]"
-	else if (total == 2)
-		return "[input[1]][and_text][input[2]]"
+	else if (uniquetotal == 1)
+		typecount = prune_list_to_type(input, types[1]).len
```
Comment (ShiftyRail) [inline code/__HELPERS/lists.dm]: This is a dangerous usage of Byond's permissive syntax and compilation rules. There is a reason why we have a check in our linter against this.
You need to put a list variable for the proc and then use the `.len` attribute of this var.
```dm
var/list/L = prune_list_to_type(input, types[1])
typecount = L.len
```
And so on.

**Review by ShiftyRail — CHANGES_REQUESTED** [review]

**Review by ShiftyRail — COMMENTED** [review]
You may also need to use a `switch` construct rather than 3 if-else blocks here.

**Comment by Eneocho** [comment]
> It contains: 2 multitools, 2 jackbootss and a oxygen tank" (or "an oxygen tank", hopefully)

Can it be tweaked so it says 2 [pairs of] jackboots?

**Comment by Exxion** [comment]
Not feasibly, no.
Also I would strongly recommend separating by name instead of type.

File: code/__HELPERS/lists.dm (hunk 1 above)
Comment (ghost) [reply code/__HELPERS/lists.dm]: Done

**Comment by ghost** [comment]
> Not feasibly, no.
> Also I would strongly recommend separating by name instead of type.

This has now been done

**Comment by kane-f** [comment]
It seems like english_list() definitely isn't just used with atoms, don't merge this for now until I bring the old function back and use it in seperate cases

**Comment by kane-f** [comment]
Old function is now back, downside is nothing is really changed now and new function remains unused, will implement in certain cases later

**Review by ShiftyRail — APPROVED** [review]

**Pull request merged**